lib/rdf-borsh/src/parse.rs

```diff
@@ -0,0 +1,155 @@
+// This is free and unencumbered software released into the public domain.
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use borsh::io::Read;
+use lz4_flex::frame::FrameDecoder;
+use rdf_model::HeapTerm;
+use winnow::{
+    binary::{le_u16, le_u32, length_take, u8},
+    combinator::{dispatch, fail},
+    error::{ContextError, ErrMode, StrContext, StrContextValue},
+    PResult, Parser,
+};
+
+use crate::{BorshQuad, BorshTerm};
+
+#[derive(Debug)]
+pub enum ParseError {
+    Io(borsh::io::Error),
+    Parse(ErrMode<ContextError>),
+}
+
+pub fn parse_dataset(
+    input: &mut &[u8],
+) -> Result<(Vec<BorshTerm>, Vec<BorshQuad<u16>>), ParseError> {
+    let _quad_count = parse_header(input).map_err(ParseError::Parse)?;
+
+    let mut decompressor = FrameDecoder::new(input);
+    let mut buf = Vec::new();
+    decompressor.read_to_end(&mut buf).map_err(ParseError::Io)?;
+
+    let mut input = buf.as_slice();
+    let terms = parse_term_dictionary(&mut input).map_err(ParseError::Parse)?;
+    let quads = parse_quads_section(&mut input).map_err(ParseError::Parse)?;
+
+    Ok((terms, quads))
+}
+
+pub fn parse_header(input: &mut &[u8]) -> PResult<u32> {
+    b"RDFB"
+        .context(StrContext::Label("magic number"))
+        .context(StrContext::Expected(StrContextValue::StringLiteral("RDFB")))
+        .parse_next(input)?;
+    b'\x01'
```

Comment:
The version number should be `'1'` (ASCII 0x31), not 0x01. Not your fault, the autogenerated spec in the RDF.rb repo doesn't match the actual implementation on this, I'll need to update it.